core/trino-main/src/main/java/io/trino/metadata/DiscoveryNodeManager.java

```diff
@@ -229,55 +237,62 @@ private synchronized void refreshNodesInternal()
                 InternalNode node = new InternalNode(service.getNodeId(), uri, nodeVersion, coordinator);
                 NodeState nodeState = getNodeState(node);
 
-                switch (nodeState) {
-                    case ACTIVE:
-                        activeNodesBuilder.add(node);
-                        if (coordinator) {
-                            coordinatorsBuilder.add(node);
-                        }
+                // Once a worker is requested to be decommissioned from coordinator,
```

Comment:
Maybe 
// worker for which decommissioning was requested may still report ACTIVE state for a while

Reply:
Actually, why do we need this mechanism. What is wrong with a short lag here?

Reply:
coordinator contains the authoritative list of nodes to be decommissioned. While workers node have been notified for the state change, the logic here will immediately stop scheduling new tasks to the nodes to be decommissioned. so it ensure better that while a worker waits for its running tasks to complete, there is no new tasks will be scheduled on it.